### PR TITLE
honor idp discovery feature for all other paths

### PR DESCRIPTION
**Pull request by haishengwu-okta**
- don't whitelist for primary auth v.s. idp discovery auth but only relies on the feature to determine whether run idp discovery flow or primary auth flow.
- therefore shift the logic to backend to set up the auth flow.
- video: https://okta.box.com/s/l640apxbvql99s9h7dqb8qfs16nvk27i
- OKTA-163556

@mauriciocastillosilva-okta @yuliu-okta 
^ @hor-kanchan-okta

### CODE REVIEW

File: test/unit/spec/LoginRouter_spec.js [hunk 1]
```diff
@@ -383,11 +383,47 @@ function (Okta, Q, Backbone, SharedUtil, CryptoUtil, Logger, OktaAuth, Util, Exp
...
-        test.router.navigate('login/default');
+        test.router.navigate('/app/salesforce/abc123sef/sso/saml');
+        return Expect.waitForIDPDiscovery();
+      })
+      .then(function () {
+        var form = new IDPDiscoveryForm($sandbox);
+        expect(form.isIDPDiscovery()).toBe(true);
+      });
+    });
+    itp('navigates to IDPDiscovery for /app/salesforce/{id}/sso/saml when features.idpDiscovery is false', function () {
```
Comment (yuliu-okta) [inline test/unit/spec/LoginRouter_spec.js]: navigates to primaryAuth

File: test/unit/spec/LoginRouter_spec.js [hunk 2]
```diff
@@ -383,11 +383,47 @@ function (Okta, Q, Backbone, SharedUtil, CryptoUtil, Logger, OktaAuth, Util, Exp
...
+        Util.mockRouterNavigate(test.router);
+        test.router.navigate('/app/salesforce/abc123sef/sso/saml');
+        return Expect.waitForPrimaryAuth();
+      })
+      .then(function () {
+        var form = new PrimaryAuthForm($sandbox);
+        expect(form.isPrimaryAuth()).toBe(true);
+      });
+    });
+    itp('navigates to PrimaryAuth for /any/other when features.idpDiscovery is true', function () {
```
Comment (yuliu-okta) [inline test/unit/spec/LoginRouter_spec.js]: navigates to IdP Discovery

**Comment by haishengwu-okta** [comment]
@mauriciocastillosilva-okta 
thanks for bring it up and that's exactly we want. (unless we missed any cases)

- this is how the [feature variable set from jsp](https://github.com/okta/okta-core/blob/master/webapp/src/main/resources/META-INF/resources/WEB-INF/jsp/loginpage_backbone.jsp#L377-L380) and essentially it comes [from controller](https://github.com/okta/okta-core/blob/master/server/ui/src/main/java/com/saasure/controller/enduser/login/LoginController.java#L337)
- any url path will hit corresponding controller first and it does imply the dependency on backend to set the feature variable properly if want to support idp discovery for that url.
- basically we shift whitelist/blacklist logic from widget to backend, which I think it's better because `features.idpDiscovery` is single source of truth to enable idp discovery for widget.

**Comment by haishengwu-okta** [comment]
@yuliu-okta 
widget is a SPA hence the initial value will retain while navigation unless refresh/redirect.
